servers/zms/src/test/java/com/yahoo/athenz/zms/ZMSImplTest.java

```diff
@@ -6564,7 +6569,134 @@ public void testGetTenantResourceGroupRoles() {
+        zmsImpl.putTenantResourceGroupRoles(serviceProviderCtx, domain, serviceName, tenantDomain, resourceGroup,
+                zmsTestInitializer.getAuditRef(), tenantRoles);
+
+        TenantResourceGroupRoles tRoles = zmsImpl.getTenantResourceGroupRoles(zmsTestInitializer.getMockDomRsrcCtx(), domain, serviceName,
+                tenantDomain, resourceGroup);
+        assertNotNull(tRoles);
+        assertEquals(domain.toLowerCase(), tRoles.getDomain());
+        assertEquals(serviceName.toLowerCase(), tRoles.getService());
+        assertEquals(tenantDomain.toLowerCase(), tRoles.getTenant());
+        assertEquals(resourceGroup.toLowerCase(), tRoles.getResourceGroup());
+        assertEquals(ZMSTestInitializer.TABLE_PROVIDER_ROLE_ACTIONS.size(), tRoles.getRoles().size());
+
+        // Verify dependency
+
+        DomainList dependentDomainList = zmsImpl.getDependentDomainList(zmsTestInitializer.getMockDomRsrcCtx(), domain + "." + serviceName);
+        assertEquals(dependentDomainList.getNames().size(), 1);
+        assertEquals(dependentDomainList.getNames().get(0), tenantDomain.toLowerCase());
+
+        // Now add another resource group
+
+        String resourceGroup2 = "Group2";
+        tenantRoles = new TenantResourceGroupRoles().setDomain(domain)
+                .setService(serviceName).setTenant(tenantDomain)
+                .setRoles(roleActions).setResourceGroup(resourceGroup2);
+        zmsImpl.putTenantResourceGroupRoles(serviceProviderCtx, domain, serviceName, tenantDomain, resourceGroup2,
+                zmsTestInitializer.getAuditRef(), tenantRoles);
+
+        tRoles = zmsImpl.getTenantResourceGroupRoles(zmsTestInitializer.getMockDomRsrcCtx(), domain, serviceName,
+                tenantDomain, resourceGroup2);
+        assertNotNull(tRoles);
+        assertEquals(domain.toLowerCase(), tRoles.getDomain());
+        assertEquals(serviceName.toLowerCase(), tRoles.getService());
+        assertEquals(tenantDomain.toLowerCase(), tRoles.getTenant());
+        assertEquals(resourceGroup2.toLowerCase(), tRoles.getResourceGroup());
+        assertEquals(ZMSTestInitializer.TABLE_PROVIDER_ROLE_ACTIONS.size(), tRoles.getRoles().size());
+
+        // Now remove the first resource group and verify dependency remains until the second resource group and admin role is removed
+
+        zmsImpl.deleteTenantResourceGroupRoles(serviceProviderCtx, domain, serviceName, tenantDomain, resourceGroup, zmsTestInitializer.getAuditRef());
+        dependentDomainList = zmsImpl.getDependentDomainList(zmsTestInitializer.getMockDomRsrcCtx(), domain + "." + serviceName);
+        assertEquals(dependentDomainList.getNames().size(), 1);
+        assertEquals(dependentDomainList.getNames().get(0), tenantDomain.toLowerCase());
+
+        // Now delete the second resouce group and verify dependency still remains until the admin role is removed
```

Comment:
@havetisyan Please note that putTenantResourceGroupRoles and deleteTenantResourceGroupRoles are not symmetrical in their creation and deletion of roles.
An admin role is created in putTenantResourceGroupRoles the first time that a resource group is created for a domain. It isn't removed when deleteTenantResourceGroupRoles is called.
To actually remove the role, we use the "deleteTenant" call. 
For domain dependency - I keep the dependency until all roles are removed, including the admin. Let me know if you think the behavior should be different.